packages/angular_devkit/build_angular/src/builders/karma/find-tests-plugin.ts

```diff
@@ -6,15 +6,76 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import assert from 'assert';
 import { PathLike, constants, promises as fs } from 'fs';
 import glob, { hasMagic } from 'glob';
 import { basename, dirname, extname, join, relative } from 'path';
 import { promisify } from 'util';
+import type { Compilation, Compiler } from 'webpack';
+import { addError } from '../../utils/webpack-diagnostics';
 
 const globPromise = promisify(glob);
 
+/**
+ * The name of the plugin provided to Webpack when tapping Webpack compiler hooks.
+ */
+const PLUGIN_NAME = 'angular-find-tests-plugin';
+
+export interface FindTestsPluginOptions {
+  include?: string[];
+  workspaceRoot: string;
+  projectSourceRoot: string;
+}
+
+export class FindTestsPlugin {
+  private compilation: Compilation | undefined;
+
+  constructor(private options: FindTestsPluginOptions) {}
+
+  apply(compiler: Compiler): void {
+    const { include = ['**/*.spec.ts'], projectSourceRoot, workspaceRoot } = this.options;
+
+    compiler.hooks.environment.tap(PLUGIN_NAME, () => {
```

Comment:
Is this hook needed?  Could the compiler options be modified directly in the apply function?

Reply:
I believe so, will update.